mlflow/tracking/fluent.py

```diff
@@ -161,7 +161,10 @@ def end_run(status=RunStatus.to_string(RunStatus.FINISHED)):
     """End an active MLflow run (if there is one)."""
     global _active_run_stack
     if len(_active_run_stack) > 0:
-        MlflowClient().set_terminated(_active_run_stack[-1].info.run_id, status)
```

Comment:
Could we just reorder the lines appropriately so that we call the set_terminated after unsetting the global state? It seems OK to re-throw the exception, but the run should be ended afterwards.

Reply:
The problem I was running into was when the files related to the run don't exist anymore, resulting in a deadlock, where I can't start a new run or end the run mlflow thinks is active.

Reply:
in retrospective, it might be better idea to handle this in the `file_store`?

Reply:
Can you catch a more specific Exception like FileNotFound rather than a wider MLflowException?

Reply:
The problem I ran into comes from: https://github.com/mlflow/mlflow/blob/7c9ef36c338fb6eb33d8b9f5eb23b9ca556731a9/mlflow/store/tracking/file_store.py#L442

So I am passing over only that case